packages/backend/src/models/UserWarehouseCredentials/UserWarehouseCredentialsModel.ts

```diff
@@ -176,11 +177,33 @@ export class UserWarehouseCredentialsModel {
             userUuid,
             warehouseType,
         );
+
         if (credentials) {
-            return this.convertToUserWarehouseCredentialsWithSecrets(
-                credentials,
-            );
+            const credentialsWithSecrets =
+                this.convertToUserWarehouseCredentialsWithSecrets(credentials);
+
+            // Fixes backwards compatibility token for snowflake
+            // Without this, refreshCredentials will not have the refreshToken
+            // So it will try to fetch the refresh token from the user's OpenID table
+            // Instead of the user warehouse credentials table
+            if (
+                credentialsWithSecrets.credentials.type ===
+                WarehouseTypes.SNOWFLAKE
+            ) {
+                if (!credentialsWithSecrets.credentials.refreshToken) {
+                    throw new SnowflakeTokenError(
```

Comment:
This will throw an error and force users to reauthenticate if snowflake user credentials is using `token` instead of `refreshToken` 

<img width="1964" height="467" alt="Image" src="https://github.com/user-attachments/assets/57b9a4cf-f045-4bc9-8c91-a08e050a8100" />

Reply:
Updated

<img width="799" height="237" alt="Image" src="https://github.com/user-attachments/assets/67a01a2e-1392-4fd7-91c0-59c9b20798f3" />

<img width="1825" height="396" alt="Image" src="https://github.com/user-attachments/assets/000ac0ae-4652-4b9b-b865-fff90732dfa3" />